camayoc/tests/qpc/api/v1/reports/test_reports.py

```diff
@@ -164,19 +164,17 @@ def test_OS_found_deployment_report(scans, scan_name):
         # contained in the found name.
         # For example, if "Red Hat" is listed in config file,
         # It will pass if "Red Hat Enterprise Linux Server" is found
-        if expected_distribution.name not in found_distro:
+        if found_distro.startswith(expected_distribution.name) is False:
             errors_found.append(
                 "Expected OS named {0} for source {1} but " "found OS named {2}".format(
                     expected_distribution.name,
                     hostname,
                     found_distro,
                 )
             )
-        # We assert that the expected distro's version is at least
-        # contained in the found version.
-        # For example, if "6.9" is listed in config file,
-        # It will pass if "6.9 (Santiago)" is found
-        if expected_distribution.version not in found_version:
+        # We assert that the expected distro's version is exactly
+        # the same contained in the found version.
+        if expected_distribution.version != found_version:
```

Comment:
optional: I'm not the one to complain about stronger assertions, but I'm curious **why** we need to make that change, and why _now_.

Reply:
Why do we need it? Because of a case like this:

```python
>>> expected_ver = '8.1'
>>> found_ver = '8.10'
>>> print(expected_ver not in found_ver)
False
>>> print(expected_ver != found_ver)
True
```
Why now? The current code is not correct.

Reply:
Ah, alright, that makes sense :+1: